src/daemon/distribution.c

```diff
@@ -0,0 +1,148 @@
+        }
+        else 
+        {
+            new_distribution->buckets[i] = initialize_bucket(new_distribution->buckets[i - 1].max_boundary + 1, INFINITY);
+        }
+    }
+    
+    new_distribution->num_buckets = num_buckets;
+    new_distribution->total_scalar_count = 0;
+    new_distribution->raw_data_sum = 0;
+}
+
+distribution_t * distribution_new_exponential(size_t num_buckets, double initial_size, double factor) {
+    distribution_t *new_distribution = calloc(1, sizeof(distribution_t));
+    new_distribution->buckets = calloc(num_buckets, sizeof(bucket_t));
+    double previous_bucket_size;
+    double new_max;
+
+    for(int i = 0; i < num_buckets; i++)
+    {
+        if(i == 0)
+        {
+            new_distribution->buckets[i] = initialize_bucket(0, initial_size);
+        }
+        else if(i < num_buckets - 1)
+        {
+            previous_bucket_size = new_distribution->buckets[i - 1].max_boundary - new_distribution->buckets[i - 1].min_boundary;
+            new_max = new_distribution->buckets[i - 1].max_boundary + previous_bucket_size * factor;
+            new_distribution->buckets[i] = initialize_bucket(new_distribution->buckets[i - 1].max_boundary + 1, new_max);
+        }
+        else
+        {
+            new_distribution->buckets[i] = initialize_bucket(new_distribution->buckets[i - 1].max_boundary + 1, INFINITY);
+        }
+        
+    }
+    
+    new_distribution->num_buckets = num_buckets;
+    new_distribution->total_scalar_count = 0;
+    new_distribution->raw_data_sum = 0;
+}
+
+distribution_t * distribution_new_custom(size_t num_buckets, double *custom_buckets_sizes) {
+    distribution_t *new_distribution = calloc(1, sizeof(distribution_t));
+    new_distribution->buckets = calloc(num_buckets, sizeof(bucket_t));
+
+    for(size_t i = 0; i < num_buckets; i++)
+    {
+        if(i == 0)
+        {
+            new_distribution->buckets[i] = initialize_bucket(0, custom_buckets_sizes[0]);
+        }
+        else if(i < num_buckets - 1)
+        {
+            new_distribution->buckets[i] = initialize_bucket(new_distribution->buckets[i - 1].max_boundary + 1, custom_buckets_sizes[i]);
+        }
+        else
+        {
+            new_distribution->buckets[i] = initialize_bucket(new_distribution->buckets[i - 1].max_boundary + 1, INFINITY);
+        }
+    }
+    
+    new_distribution->num_buckets = num_buckets;
+    new_distribution->total_scalar_count = 0;
+    new_distribution->raw_data_sum = 0;
+}
+
+void distribution_update(distribution_t *dist, double gauge) {
+    for(size_t i = 0; i < dist->num_buckets; i++)
+    {
+        if(gauge >= dist->buckets[i].min_boundary && gauge <= dist->buckets[i].max_boundary)
+        {
+            dist->buckets[i].bucket_counter++;
+        }
+    }
+}
+
+double distribution_average(distribution_t *dist) {
+    return dist->raw_data_sum / dist->total_scalar_count;
+}
+
+int distribution_percentile(distribution_t *dist, uint8_t percent) {
```

Comment:
`percent` should be a `double`, so that users can provide a value such as `99.5`.

Reply:
Okay, I changed it.